Update of traditional CFDI 3.3 to 4.0

NotMorales opened the pull request:
- [x] Added new 'Global Information' node
- [x] Added new attribute 'Export'
- [x] Added the new attribute 'Domicilio fiscal receptor' and 'Régimen fiscal receptor' to the 'Receptor' 
- [x] Update of the 'Conceptos' Node
- [x] Added relevant CFDI catalogs

NotMorales committed:
Update CFDI 3.3 to CFDI 4.0

NotMorales committed:
Adjustment of the tax table

NotMorales committed:
Added support for the "Pagos20" plugin

eclipxe13 committed:
Fix build process

- code style
- test suite changed to PHPUnit 9.5
- remove test/PdtToText/ShellExec (CfdiUtils) dependency, use Symfony Process Component

shnere commented on templates/generic.php:
```diff
@@ -312,6 +364,260 @@
+                        <strong>Moneda DR: </strong><span><?=$this->e($doctoRelacionado['MonedaDR'])?></span>
+                        <strong>Equivalencia DR: </strong>
+                            <span><?=$this->e($doctoRelacionado['EquivalenciaDR'])?></span>
+                        <strong>Tipo de cambio DR: </strong>
```
Not needed, it is superseeded by `EquivalenciaDR`.

eclipxe13 replied:
Fixed at 2022-09-22

shnere commented on templates/generic.php:
```diff
@@ -312,6 +364,260 @@
+                            <span><?=$this->e($doctoRelacionado['EquivalenciaDR'])?></span>
+                        <strong>Tipo de cambio DR: </strong>
+                            <span><?=$this->e($doctoRelacionado['TipoCambioDR'])?></span>
+                        <strong>Método de pago DR: </strong>
```
Not needed either.

eclipxe13 replied:
Fixed at 2022-09-22

shnere commented on templates/generic.php:
```diff
@@ -15,11 +16,14 @@
 $pagoCount = $pagos->count();
-
+if (! isset($catalogos)) {
+    $catalagos = new \PhpCfdi\CfdiToPdf\CfdiCatalogs();
```
```suggestion
    $catalogos = new \PhpCfdi\CfdiToPdf\CfdiCatalogs();
```

The typo is present in multiple places throughout this file.

eclipxe13 replied:
Fixed at 2022-09-22

shnere commented on templates/generic.php:
```diff
@@ -312,6 +364,260 @@
+                            <strong>Impuestos Docto Relacionado</strong>
+                        </p>
+                            <?php
+                            $traslados = $impuestos->searchNodes('pago20:RetencionesDR', 'pago20:RetencionDR');
+                            $retenciones = $impuestos->searchNodes('pago20:TrasladosDR', 'pago20:TrasladoDR');
```
Variables are mixed up, causing issues when representing those values:
```suggestion
                            $retenciones = $impuestos->searchNodes('pago20:RetencionesDR', 'pago20:RetencionDR');
                            $traslados = $impuestos->searchNodes('pago20:TrasladosDR', 'pago20:TrasladoDR');
```

shnere commented on templates/generic.php:
```diff
@@ -312,6 +364,260 @@
+                        <strong>Impuestos Pago</strong>
+                    </p>
+                    <?php
+                    $traslados = $impuestos->searchNodes('pago20:RetencionesP', 'pago20:RetencionP');
+                    $retenciones = $impuestos->searchNodes('pago20:TrasladosP', 'pago20:TrasladoP');
```
```suggestion
                    $retenciones = $impuestos->searchNodes('pago20:RetencionesP', 'pago20:RetencionP');
                    $traslados = $impuestos->searchNodes('pago20:TrasladosP', 'pago20:TrasladoP');
```

eclipxe13 replied:
Fixed at 2022-09-22

shnere commented on templates/generic.php:
```diff
@@ -312,6 +364,260 @@
+    if (null !== $pagoTotales) :
+        ?>
+        <div class="panel">
+            <div class="title">Pagos</div>
```
`Complemento Pago` may be clearer.

eclipxe13 replied:
Fixed at 2022-09-22

shnere commented on templates/generic.php:
```diff
@@ -312,6 +364,260 @@
+                    <?php endforeach; ?>
+                <?php endif; ?>
+                <?php $impuestos = $pago20->searchNode('pago20:ImpuestosP');
+                if (null !== $impuestos) : ?>
```
It would be clearer if we move upwards this section of Impuestos Pago along with the Pago.

eclipxe13 replied:
Fixed at 2022-09-22

eclipxe13 committed:
Upgrade development tools

- Code style to PSR-12
- Update PHPUnit config file
- Update PHPStan config file

eclipxe13 committed:
Use phpcfdi/cfdi-cleaner instead of CfdiUtils Cleaner

eclipxe13 committed:
Bump to PHP 7.3

- Solve some PHPStan typo issues

eclipxe13 committed:
Refactor PdfToText

eclipxe13 committed:
Improve readability

- Import classes
- Fix typos

eclipxe13 committed:
Refactor generic template

- set $impuestos just before use
- fix retenciones and traslados variable allocation
- remove pagos20 TipoCambioDR (it does not exists)

eclipxe13 committed:
Introduce CatalogsInterface and moce CfdiCatalogs to StaticCatalogs

- On generic template, catalogs can be introduced using a different class that implements CatalogsInterface

eclipxe13 committed:
Run phpunit using the same php command as composer was invoked

eclipxe13 committed:
Remove test/_files/external-resources/

eclipxe13 committed:
Introduce development tool composer-normalize

eclipxe13 committed:
Move from Travis-CI to GH Workflows

eclipxe13 committed:
Update project description and typos

eclipxe13 committed:
Improve Pagos 20

- Remove non existent MetodoDePagoDR
- Move "Totales del complemento de pago" after pagos

eclipxe13 committed:
Short argument for version is -V (uppercase)

eclipxe13 committed:
Add fonts directory option, extract from phar tcpdf fonts

eclipxe13 committed:
Migrate from phar-builder to humbug/box

eclipxe13 committed:
Fix typos and titles

eclipxe13 committed:
Prepare version 0.4.0
